### ISSUE TITLE
Shapes get stuck into polygon edges.

### ISSUE THREAD

**Issue by yosoyfreeman** [godot-jolt/godot-jolt#508]
I been using Godot jolt for a few days and today i noticed that, contrary to what is indicated [here](https://www.youtube.com/watch?v=EanFxlkZgcA) shapes do not ignore their internal edges.  This is specially apparent while using cylinders, but i used a box to simplify the visualization. 

**Godot physics**:
![Godot physics](https://github.com/godot-jolt/godot-jolt/assets/64601424/0f2584d0-de00-4335-ac1a-1e61aa9ff084)

**Godot Jolt:**
![Jolt](https://github.com/godot-jolt/godot-jolt/assets/64601424/a27aafa3-3348-4784-9783-8f59a8c0c8ff)

All settings are the default ones from both engines on the test, but in my experience, increasing velocity/position iterations increases the problem as it seems to detect this edges more precisely and therefore being stopped/affected more.

Reproduction project(Godot 4 Stable):
[StuckBug.zip](https://github.com/godot-jolt/godot-jolt/files/12163376/StuckBug.zip)

**Comment by mihe** [comment]
I can't speak to whether this phenomenon actually relates to active edges to begin with, but if you turn down the "Contact Speculative Distance" project setting to 0 this issue seems to go away, so I'm not entirely sure whether to label this a bug or not.

I'd be curious to hear your thoughts, @jrouwe. Here's roughly the equivalent `Samples` code:

```cpp
TriangleList triangles = {
    {Float3(-2.0f, 2.0f, 2.0f), Float3(2.0f, 2.0f, 2.0f), Float3(-2.0f, 2.0f, -2.0f)},
    {Float3(-2.0f, 2.0f, -2.0f), Float3(2.0f, 2.0f, 2.0f), Float3(2.0f, 2.0f, -2.0f)},

    {Float3(-2.0f, 2.0f, -2.0f), Float3(2.0f, 2.0f, -2.0f), Float3(-2.0f, 0.0f, -4.0f)},
    {Float3(-2.0f, 0.0f, -4.0f), Float3(2.0f, 2.0f, -2.0f), Float3(2.0f, 0.0f, -4.0f)},
};

BodyCreationSettings box(new BoxShape(Vec3(1, 1, 1)), RVec3(0, 3, 0), Quat::sIdentity(), EMotionType::Dynamic, Layers::MOVING);
box.mLinearVelocity = Vec3(0, 0, -0.5f);
box.mAngularDamping = 0.1f;
box.mLinearDamping = 0.1f;
box.mFriction = 0.0f;

mBodyInterface->CreateAndAddBody(box, EActivation::Activate);
mBodyInterface->CreateAndAddBody(BodyCreationSettings(new MeshShapeSettings(triangles), RVec3::sZero(), Quat::sIdentity(), EMotionType::Static, Layers::NON_MOVING), EActivation::DontActivate);

PhysicsSettings physics_settings = mPhysicsSystem->GetPhysicsSettings();
// physics_settings.mSpeculativeContactDistance = 0.0f;
mPhysicsSystem->SetPhysicsSettings(physics_settings);
```

**Comment by yosoyfreeman** [comment]
>  Thank you for reporting this, and for providing the minimal reproduction project.

Thank you for taking the time of looking into it. I would recommend making the minimal reproduction project part of the reporting bug template as Godot does, i think is going to make your work lot easier, specially to see if old bugs are still reproducible in new versions without repeating work.

**Comment by mihe** [comment]
> I would recommend making the minimal reproduction project part of the reporting bug template as Godot does

I've gone back and forth on whether I should add an issue template or not. I don't really get the kinds of volumes where it's any burden for me to just ask for an MRP if it's missing, and I'd prefer to have as little friction as possible for people wanting to report an issue, seeing as how there's so few of them right now.

**Comment by yosoyfreeman** [comment]
I tested this more in deep and noticed that while setting speculative contacts to 0 prevent the issue on the showed minimal reproduction project, it still produces jumps over the edge, so its still affecting the body (sometimes even more) under some conditions. Please, could you tell me how big reproduction projects should be at max to not cause you troubles? so i can eliminate enough to be minimal but keep enough so you can move around a little and see the inconsistency of the behavior.

**Comment by mihe** [comment]
> while setting speculative contacts to 0 prevent the issue on the showed minimal reproduction project, it still produces jumps over the edge

If you still find yourself getting stuck on edges, like in your original MRP, then you might want to experiment with lowering "Contact Allowed Penetration" as well, which is supposed to be okay to bring down to 0 as well, although I have seen some artifacts when doing so myself.

> Please, could you tell me how big reproduction projects should be at max to not cause you troubles?

I'll assume you're talking about the scope of the project and not its actual file size. That's a tricky thing to quantify, but I appreciate reproduction projects where you see the problem as soon as you run the main scene, and where the problematic behavior isn't dependent on hundreds of lines of some custom character controller, since they more often than not turn out to be the source of the problem. If you absolutely need a `CharacterBody3D` then try to stick to the default template, if possible.

**Comment by yosoyfreeman** [comment]
Thank you! I was talking in fact about file size. I know i must keep reproduction issues simple in order for them to be useful.

**Comment by mihe** [comment]
Oh, alright, file size isn't a concern. GitHub will limit you at something like 10 MB either way, I think.

**Comment by mihe** [comment]
@yosoyfreeman

I've just published [0.6.0-rc1](https://github.com/godot-jolt/godot-jolt/releases/tag/v0.6.0-rc1), which includes the change to the active/inactive threshold angle. I'd appreciate it if you could give it a try and confirm that it does indeed resolve this issue.

**Comment by yosoyfreeman** [comment]
Sure! i have time in the afternoon, i will edit this post.
